srcs/app/http/HttpRequest.cpp

```diff
@@ -188,10 +260,71 @@ bool	HttpRequest::ContainOnlyVisibleChars_(const std::string &str) const {
 	return true;
 }
 
+// Parse the Host header into a host and optional port number
+// Host = uri-host [ ":" port ]
+bool	HttpRequest::ParseHost_() {
+	std::string host = GetHeaderValue("Host");
+	if (host.empty())
+		return false;
+	std::size_t port_delimiter = host.find(':');
+	if (port_delimiter == std::string::npos) {
+		host_ = host;
+		port_ = 80;
```

Comment:
Watch out with this because if serving through https port would be 443! Another comment I want to make is if it makes sense to parse the port since the OS will be responsible for redirecting requests to the right port/socket

Reply:
Good point! No idea if this is required when making an HTTP request from a browser. I can think of a case when using a custom client, let's say the client connects to IP:port then sends a request where the Host header has a different port.